prototype_source/backend_config_tutorial.rst

```diff
@@ -0,0 +1,447 @@
+------------------
+
+Here we define a simple example BackendConfig that supports 3 things:
+
+- Quantizing linear to int8 (unsigned)
+- Fusing conv2d followed by relu
+- Quantizing the fused conv2d-relu to int8 (unsigned)
+
+In this section, we will run through an example model with these
+patterns and show how to use this BackendConfig in ``prepare_fx`` and
+``convert_fx``.
+
+.. code:: ipython3
+
+    import torch
+    from torch.ao.quantization import (
+        default_weight_observer,
+        get_default_qconfig_mapping,
+        MinMaxObserver,
+        QConfig,
+        QConfigMapping,
+    )
+    from torch.ao.quantization.backend_config import (
+        BackendConfig,
+        BackendPatternConfig,
+        DTypeConfig,
+        DTypeWithConstraints,
+        ObservationType,
+    )
+    from torch.ao.quantization.quantize_fx import prepare_fx, convert_fx
+
+.. code:: ipython3
+
+    # ======================
+    #  Custom BackendConfig
+    # ======================
+    
+    quint8_with_constraints = DTypeWithConstraints(
+        dtype=torch.quint8,
+        quant_min_lower_bound=0,
+        quant_max_upper_bound=255,
+        scale_min_lower_bound=2 ** -12,
+    )
+    
+    # Specify the dtypes passed to the quantized ops in the reference model spec
+    # See the "Data Type Restrictions" section above for more detail
+    weighted_int8_dtype_config = DTypeConfig(
+        input_dtype=quint8_with_constraints,
+        output_dtype=quint8_with_constraints,
+        weight_dtype=torch.qint8,
+        bias_dtype=torch.float)
+
+   def fuse_conv2d_relu(is_qat, conv, relu):
+       """Return a fused ConvReLU2d from individual conv and relu modules."""
+       return torch.ao.nn.intrinsic.ConvReLU2d(conv, relu)
+    
+    # The following BackendPatternConfig refers to static quantization for the
+    # linear op. This inserts quant-dequant ops before and after the float linear
+    # op, producing the following reference model spec:
+    #
+    #   quant1 - [dequant1 - fp32_linear - quant2] - dequant2
+    #
+    # Here we specify using different observers (will be renamed) for the input
+    # and output for the linear op, so the quantization params passed to the two
+    # quantize ops will be different. This is commonly the case for weighted ops
+    # like linear and conv.
+    #
+    # The input dtype specified in the DTypeConfig will be passed as the dtype
+    # argument to quant1, while the output dtype will be passed as the dtype
+    # argument to quant2. If the output dtype is fp32, as in the case of dynamic
+    # quantization, then the output quant-dequant pair will not be inserted.
+    #
+    # Ultimately, how this reference model spec will be fused will depend on
+    # how the backend lowers the model. For static quantization, for example,
+    # the backend may fuse the reference pattern in the square brackets into
+    # a single quantized linear op.
+
+    linear_config = BackendPatternConfig() \
```

Comment:
another thing I want to mention is that we can structure the 3 configs to linear and conv_relu, something like:

let's say the target backend has two operators: quantized::linear and quantized::conv2d_relu, 
1. here is the reference pattern for quantized::linear, 
2. here are the steps we'll get this pattern
3. here is the BackendPatternConfig

and do the same for quantized::conv2d_relu op

This would help backend developers to get the right mental model of how they can come up with the backend pattern config for the target quantized operator I think